### PR TITLE
Add retrieve command for cold storage logs

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: cmd/oklog/retrieve.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"storj.io/uplink"
+)
+
+func runRetrieve(args []string) error {
+	flagset := flag.NewFlagSet("retrieve", flag.ExitOnError)
+	var (
+		bucketName  = flagset.String("bucket", "", "bucket name to retrieve logs from")
+		accessGrant = flagset.String("access", os.Getenv("ACCESS_GRANT"), "access grant from satellite")
```
Comment (kristaxox) [inline cmd/oklog/retrieve.go]: Wouldn’t this access grant already exist in the context of the application? For pushing?

Comment (kristaxox) [reply cmd/oklog/retrieve.go]: I'm assuming the retrieve command still hits the oklog API endpoint?

Comment (halkyon) [reply cmd/oklog/retrieve.go]: @kristaxox Yes, absolutely right, I still need to shift the uplink stuff into an API endpoint (so we don't have to duplicate the access and bucket flags) and have this command just query that :)

**Review by kristaxox — COMMENTED** [review]
Looks great so far!

**Comment by halkyon** [comment]
Have moved the uplink stuff into `fileLogDCS`, with the retrieve command being an API client that pulls data from the store API. The diff here is against Artur's `add-fileLogDCS` branch, once that's merged I'll re-target to `main`.

Still need to tidy up things and look at some tests, but I'm getting happier with this :)

File: cmd/oklog/retrieve.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	case durationErr == nil && timeErr != nil:
+		toStr = time.Now().Add(neg(toDuration)).Format(time.RFC3339)
+	case durationErr != nil && timeErr == nil:
+		toStr = toTime.Format(time.RFC3339)
+	default:
+		return fmt.Errorf("couldn't parse -to (%q) as either duration or time", *to)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf(
+		"http://%s/store%s?from=%s&to=%s",
```
Comment (kristaxox) [inline cmd/oklog/retrieve.go:75]: Is this the correct path?

Comment (halkyon) [reply cmd/oklog/retrieve.go:75]: yep!

Comment (kristaxox) [reply cmd/oklog/retrieve.go:75]: Oh sorry! I totally thought I was looking at a test_retrieve.go type file 😂 oops!

**Review by kristaxox — APPROVED** [review]

File: cmd/oklog/retrieve.go (hunk 2 above)
Comment (halkyon) [reply cmd/oklog/retrieve.go:75]: good point though, there aren't any tests 😳 let's see if I can get at least one included!

**Comment by halkyon** [comment]
Major bits missing are tests and sorting of the segments coming from the bucket. I'm looking at tests now, and I've added a todo in the code for sorting to be done later!

I did notice the existing query functionality uses `fs.Filesystem` which can be swapped out with a virtual one for tests. An equivalent filesystem for uplink may help here so we can reuse code like `filelog.Query` to retrieve logs from DCS, but out of scope for this hackathon. Really good to know the codebase supports this abstraction though.

**Comment by kristaxox** [comment]
@halkyon not sure what I've done here :S. I merged in @amwolff 's changes and this happened automagically.

**Pull request closed without merging**